core/src/test/java/org/apache/iceberg/TestMetrics.java

```diff
@@ -766,24 +771,38 @@ protected void assertCounts(int fieldId, Long valueCount, Long nullValueCount, M
 
   protected void assertCounts(
       int fieldId, Long valueCount, Long nullValueCount, Long nanValueCount, Metrics metrics) {
-    Map<Integer, Long> valueCounts = metrics.valueCounts();
-    Map<Integer, Long> nullValueCounts = metrics.nullValueCounts();
-    Map<Integer, Long> nanValueCounts = metrics.nanValueCounts();
-    Assert.assertEquals(valueCount, valueCounts.get(fieldId));
-    Assert.assertEquals(nullValueCount, nullValueCounts.get(fieldId));
-    Assert.assertEquals(nanValueCount, nanValueCounts.get(fieldId));
+    assertThat(metrics.valueCounts())
+        .satisfiesAnyOf(
```

Comment:
I'm not sure I understand why this uses `satisfiesAnyOf`. Why not just use `containsEntry()` directly here?

Reply:
Here, in some cases, `valueCounts`, `nullValueCounts` and `nanValueCount` don't have `fieldId`, this causes an assertion error as below:

```
java.lang.AssertionError: 
Expecting map:
  {}
to contain entries:
  [1=null]
but could not find the following map entries:
  [1=null]
	at org.apache.iceberg.TestMetrics.assertCounts(TestMetrics.java:774)
	at org.apache.iceberg.TestMetrics.assertCounts(TestMetrics.java:769)
	at org.apache.iceberg.TestMetrics.testNoneMetricsMode(TestMetrics.java:566)
```

So, I change this part to check if the `fieldId` exists, or the key-value pair.

Reply:
Change this part like the previous tests. If `containsEntry` is used, it's also possible to write like:

```java
assertThat(metrics.valueCounts()).satisfies(
        valueCounts -> {
          if (valueCounts.containsKey(fieldId)) assertThat(valueCounts).containsEntry(fieldId, valueCount);
          else assertThat(valueCounts).doesNotContainKey(fieldId);
        }
);
```